src/adaXT/decision_tree/splitter.pyx

```diff
@@ -12,6 +13,41 @@ cdef double EPSILON = 2*np.finfo('double').eps
 
 cdef double INFINITY = np.inf
 
+cdef double[:] current_feature_values
+
+cdef int compare(const void* a, const void* b) noexcept nogil:
+    cdef:
+        int a1 = (<int *> a)[0]
+        int b1 = (<int *> b)[0]
+
+    if  current_feature_values[a1] >= current_feature_values[b1]:
+        return 1
+    else:
+        return -1
```

Comment:
Why -1 instead of 0?

Reply:
the q sort works with greater than, equal to or less than, which respectively would refer to 1, 0 or -1. In the equal case, the order of the two elements are undefined. I just skipped this case and included it in the greater than case (1), such that we only have the less than case left (-1). Saves us for doing a separate check of equal to and not just greater than.